### DISCUSSION TITLE
Tasmota Smart Plug  Home Assistant ESPHOME TASMOTA JMRI Lights Table

### DISCUSSION THREAD

**Discussion by trgrassijr55** [arendst/Tasmota#21852] [Q&A]
I have a model railroad in my home and I use JMRI  to run trains inventory etc.  In JMRI I have a connector for MQTT  I also run Home Assistant with ESPHOME MQTT and Tasmota  which is new to me.  I have many ESPHOME  devices all Athom Smart Plugs the YAML file has code for MQTT so they connect to my JMRI MQTT connector and then I add entries into the JMRI Lights table I set times in the JMRI Lights table and lights and devices power up and certain times of the day.  Athome no longer makes Version 2 Smart Plugs they went to Version 3 and they do not work on ESPHOME  I tried KAUF smart plugs and they failed also.   So I thought I would try Tasmota smart plugs.  The one I have is configured on my network and I can browse the device from Home Assistant. I configured the Tasmota Smart plug and I can see it in MQTT Explorer as a tasmota discovery  config and sensors.  I just need help with getting the proper syntax for the JMRI Lights table.   I tried CMND/farmland-lights  stat/farmland-lights tele/farmland-lights   none turnon the plug  I can manuly turn the plug on and off in my dashboard    Anyone know how to setup the JMRI lights table for the tasmota smart plug?
Thank you 
Tom

**Comment by sfromis** [comment]
Knowledge about JMRI here may be "less than sparse" here, but turning a plug on via Tamota would with your naming be publishing to topic `cmnd/farmland-lights/Power` with payload `on` or `off`.

**Reply by trgrassijr55** [reply]
tried cmnd/farmland-lights/Power     no luck   what do you mean by payload on or off?
I tried cmnd/farmland-lights/Power = on   no luck with that either

**Reply by sfromis** [reply]
Not knowing how JMRI works, I can't say what you need to do there, only that how MQTT works is that the basic message structure has two separate fields. First a topic which acts as an address (in this case including the command name which is "Power"), and Tasmota is subscribing to any topic starting with `cmnd/farmland-lights/` (obviously assuming that you configured Tasmota for "farmland-lights"). Then a payload which is the content of the message you send to the device(s) listening for the topic.

What you already have with ESPHome and MQTT may have clues about how MQTT is handled in JMRI, how you handle using both topic and payload.

**Reply by trgrassijr55** [reply]
When I look at MQTT and devices I do not see my Farmland Tasmota device listed. must be missing something still not sure what to look for  The device should show it that list correct?

**Reply by trgrassijr55** [reply]
Found it had to check off the filter for Tasmota to show those devices
